lib/convection/model/template/resource/aws_iam_role.rb

```diff
@@ -36,49 +37,23 @@ def trust_service(name, &block)
 
           ## Add a canned trust policy for EC2 instances
           def trust_ec2_instances(&block)
-            @trust_relationship = Model::Mixin::Policy.new(:name => 'trust-ec2-instances', :template => @template)
-            trust_relationship.allow do
-              action 'sts:AssumeRole'
-              principal :Service => 'ec2.amazonaws.com'
-            end
-            trust_relationship.instance_exec(&block) if block
-            trust_relationship
+            trust_service('ec2', 'trust-ec2-instances', &block)
           end
 
           ## Add a canned trust policy for Flow Logs
           def trust_flow_logs(&block)
-            @trust_relationship = Model::Mixin::Policy.new(:name => 'trust-flow-logs', :template => @template)
-            trust_relationship.allow do
-              action 'sts:AssumeRole'
-              principal :Service => 'vpc-flow-logs.amazonaws.com'
-            end
-            trust_relationship.instance_exec(&block) if block
-            trust_relationship
+            trust_service('vpc-flow-logs', 'trust-flow-logs', &block)
           end
 
           ## Add a canned trust policy for EMR
           def trust_emr(&block)
-            @trust_relationship = Model::Mixin::Policy.new(:name => 'trust-emr', :template => @template)
-            trust_relationship.allow do
-              action 'sts:AssumeRole'
-              principal :Service => 'elasticmapreduce.amazonaws.com'
-            end
-            trust_relationship.instance_exec(&block) if block
-            trust_relationship
+            trust_service('elasticmapreduce', 'trust-emr', &block)
+            trust_service('', '', &block)
```

Comment:
Fixed in https://github.com/rapid7/convection/pull/94/commits/d857719fbeff7a58ed16b95c2616f9d3145f937a.